src/AWS.Deploy.Orchestration/Exceptions.cs

```diff
@@ -9,7 +9,7 @@ namespace AWS.Deploy.Orchestration
     [AWSDeploymentExpectedException]
```

Comment:
All exceptions classes should have message also nullable like `Exception` base class.

Reply:
Even though the base class has the message as nullable, we do not want to allow the users to pass in a null message to our custom exceptions. This is another way to make sure exceptions are throwing valid messages to the end users as part of our UX improvement.

Reply:
I can still pass an empty string. By making it non nullable we aren't gaining anything here, it boils down to PR review.

Reply:
Right, but by having it non-nullable the intention is clear. Just because someone can work around it and pass an empty string shouldn't mean that we shouldn't be as restrictive as possible.